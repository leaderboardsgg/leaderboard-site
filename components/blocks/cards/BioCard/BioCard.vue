<template>
  <div class="bio-card">
    <CardHeader class="bio-card__header">
      <h2>About</h2>
    </CardHeader>
    <CardBody class="bio-card__content">
      <p>{{ bio }}</p>
      <div class="bio-card__buttons">
        <SocialButtons :socials="socials" />
      </div>
    </CardBody>
  </div>
</template>

<script lang="ts">
import { defineComponent } from '@nuxtjs/composition-api';
import SocialButtons from '@/components/elements/buttons/SocialButtons/SocialButtons.vue';
import CardHeader from '@/components/elements/cards/CardHeader.vue';
import CardBody from '@/components/elements/cards/CardBody.vue';

export default defineComponent({
  name: 'BioCard',
  components: { CardBody, CardHeader, SocialButtons },
  props: {
    bio: {
      default: () =>
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed eget dapibus libero. Morbi ultricies varius accumsan.',
      type: String,
    },
    socials: {
      default: () => [
        {
          icon: 'dscrd',
          name: 'Discord',
          url: 'https://discord.com/',
        },
        {
          icon: 'twch',
          name: 'Twitch',
          url: 'https://twitch.tv/',
        },
        {
          icon: 'twtr',
          name: 'Twitter',
          url: 'https://twitter.com/',
        },
        {
          icon: 'yt',
          name: 'Youtube',
          url: 'https://youtube.com/',
        },
      ],
      type: Array,
    },
  },
});
</script>

<style scoped>
.bio-card {
  @apply hidden md:flex flex-col border rounded border-gray-300;
  & > div {
    @apply p-5;
  }
  & .bio-card__header {
    @apply border-b border-gray-300;
    & h2 {
      @apply text-xl font-bold;
    }
  }
  & .bio-card__content {
    & p {
      @apply text-black;
    }
  }
  & .bio-card__buttons {
    @apply mt-5;
  }
}
</style>
