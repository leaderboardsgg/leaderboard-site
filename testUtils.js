import Vue from 'vue';
import { RouterLinkStub } from '@vue/test-utils';
import userEvent from '@testing-library/user-event';
import { render } from '@testing-library/vue';

export const NuxtLinkStub = Vue.component('NuxtLinkStub', {
  extends: RouterLinkStub,
  render(h) {
    const temp = h(
      this.tag || 'a',
      {
        attrs: {
          ...this.$vnode.data.attrs,
          ...this.$attrs,
          href: this.to,
        },
      },
      this.$slots.default,
    );

    return temp;
  },
});

export const stubbedRender = (Component, options) => {
  return render(Component, {
    stubs: { NuxtLink: NuxtLinkStub, ...options?.stubs },
    ...options,
  });
};

export * from '@testing-library/vue';
export * from '@vue/test-utils';
export { userEvent as fireEvent };
