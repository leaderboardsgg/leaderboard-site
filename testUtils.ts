import { createElementBlock, Component, ComponentOptions } from 'vue'
import { MountingOptions } from '@vue/test-utils'
import userEvent from '@testing-library/user-event'
import { render, RenderOptions } from '@testing-library/vue'

export const stubbedRender = (
  Component: Component | ComponentOptions,
  options?: RenderOptions | MountingOptions<any>,
) => {
  return render(Component, {
    global: {
      stubs: {
        NuxtLink: createElementBlock('a'),
      },
    },
    ...options,
  })
}

export * from '@testing-library/vue'
export * from '@nuxt/test-utils'
export * from '@vue/test-utils'
export { userEvent as fireEvent }
