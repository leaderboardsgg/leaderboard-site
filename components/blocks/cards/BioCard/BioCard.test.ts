import BioCard from './BioCard.vue';
import { stubbedRender } from '@/testUtils';

describe('<BioCard />', () => {
  it('should render without crashing', () => {
    const { unmount } = stubbedRender(BioCard);

    unmount();
  });

  it('renders correctly', () => {
    const { container } = stubbedRender(BioCard);

    expect(container.firstChild).toMatchSnapshot();
  });
});
