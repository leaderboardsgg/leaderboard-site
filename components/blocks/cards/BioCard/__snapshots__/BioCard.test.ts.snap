// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BioCard /> renders correctly 1`] = `
<div
  class="bio-card"
>
  <div
    class="card-header bio-card__header"
  >
    <h2>
      About
    </h2>
  </div>
   
  <div
    class="card-body bio-card__content"
  >
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed eget dapibus libero. Morbi ultricies varius accumsan.
    </p>
     
    <div
      class="bio-card__buttons"
    >
      <div
        class="social-buttons"
      >
        <a
          alt="Discord"
          class="social-button"
          href="https://discord.com/"
        >
          
    dscrd
  
        </a>
        <a
          alt="Twitch"
          class="social-button"
          href="https://twitch.tv/"
        >
          
    twch
  
        </a>
        <a
          alt="Twitter"
          class="social-button"
          href="https://twitter.com/"
        >
          
    twtr
  
        </a>
        <a
          alt="Youtube"
          class="social-button"
          href="https://youtube.com/"
        >
          
    yt
  
        </a>
      </div>
    </div>
  </div>
</div>
`;
